src/include/network/lan/complex.rb

```diff
@@ -305,13 +305,20 @@ def enableDisableButtons
 
     # Automatically configures slaves when user enslaves them into a bond or bridge device
     def UpdateSlaves
+      # TODO: sometimes it is called when no device is selected and then it crashes
+      return if LanItems.GetCurrentName.nil? || LanItems.GetCurrentName.empty?
+
       current = LanItems.current
-      master_builder = Y2Network::InterfaceConfigBuilder.for(LanItems.GetCurrentType())
+      config = Lan.yast_config.copy
+      connection_config = config.connections.find { |c| c.name == LanItems.GetCurrentName }
```

Comment:
We might think about adding a find_connection method to `Y2Network::Config`. Or to promote the array of connection configs to a "collection" (similar to InterfacesCollection).

Reply:
yes, same as with ip_configs default. Some helper would be nice. and BTW also some easy way to assign back connection_config to yast_config after modifications would be nice. As currently changes are lost as I use copy of yast_config